File: ts-lambda-worker/src/workflows.ts
import { proxyActivities } from '@temporalio/workflow';
import type * as activities from './activities';

const { greet } = proxyActivities<typeof activities>({
  startToCloseTimeout: '30 seconds',
});

export async function greetingWorkflow(name: string): Promise<string> {
  return await greet(name);
}

File: py-lambda-worker/app/workflows.py
from datetime import timedelta

from temporalio import workflow
from temporalio.common import RetryPolicy, VersioningBehavior

with workflow.unsafe.imports_passed_through():
    from .activities import GreetingInput, greet


@workflow.defn(versioning_behavior=VersioningBehavior.PINNED)
class GreetingWorkflow:
    @workflow.run
    async def run(self, input: GreetingInput) -> str:
        kwargs = dict(
            start_to_close_timeout=timedelta(seconds=input.start_to_close_seconds),
            retry_policy=RetryPolicy(maximum_attempts=input.max_attempts),
        )
        if input.heartbeat_timeout_seconds > 0:
            kwargs["heartbeat_timeout"] = timedelta(
                seconds=input.heartbeat_timeout_seconds
            )
        return await workflow.execute_activity(greet, input, **kwargs)
